src/NHibernate.Test/Associations/OneToOneFixture.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Linq;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+using NUnit.Framework;
+
+namespace NHibernate.Test.Associations.OneToOne
+{
+	[TestFixture]
+	public class OneToOneFixture: TestCaseMappingByCode
+	{
+		[Test]
+		public void OneToOneCompositeQueryByEntityParam()
+		{
+			using (var session = OpenSession())
+			{
+				var e = session.Query<EntityWithCompositeId>().FirstOrDefault();
+				var loadedEntity = session.Query<Parent>().Where(p => p.OneToOneComp == e).FirstOrDefault();
```

Comment:
Offtopic: Is there a way to see generated hql for given LINQ? I mean in some readable form not as node tree.

Reply:
No. The closest you can get is `To<Somehting>String` (I don’t remember the exact method name). It will create a human readable string for the tree.

Reply:
Did you mean this?
`nhLinqExpression.ExpressionToHqlTranslationResults.Statement.AstNode.ToStringTree()`

It gives something like: 
`( query ( select_from ( from ( range NHibernate.Test.Associations.OneToOne.Parent p ) ) ( select p ) ) ( where ( == ( . p OneToOneComp ) ( : p1 ) ) ) ( take 1 ) )`
Is that the best we can get?

Reply:
Yes, that one. Do the same with desired HQL and compare the trees. This is what I usually do.

Reply:
Ok. Thanks